Mathlib/Data/Rat/Star.lean

```diff
@@ -1,52 +1,42 @@
 /-
 Copyright (c) 2023 Jireh Loreaux. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Jireh Loreaux
+Authors: Jireh Loreaux, Yaël Dillies
 -/
+import Mathlib.Algebra.GroupWithZero.Power
 import Mathlib.Algebra.Star.Order
-import Mathlib.Data.Rat.Lemmas
-import Mathlib.Data.Rat.Order
-import Mathlib.GroupTheory.Submonoid.Membership
+import Mathlib.Algebra.Parity
 
 #align_import data.rat.star from "leanprover-community/mathlib"@"31c24aa72e7b3e5ed97a8412470e904f82b81004"
 
-/-! # Star order structure on ℚ
+/-!
+# Star ordered ring structure on ℚ
 
-Here we put the trivial `star` operation on `ℚ` for convenience and show that it is a
-`StarOrderedRing`. In particular, this means that every element of `ℚ` is a sum of squares.
+This file shows that `ℚ` is a `StarOrderedRing`. In particular, this means that every nonnegative
+rational number is a sum of squares.
 -/
 
+open AddSubmonoid Set
 
 namespace Rat
 
-instance : StarRing ℚ where
-  star := id
-  star_involutive _ := rfl
-  star_mul _ _ := mul_comm _ _
-  star_add _ _ := rfl
-
-instance : TrivialStar ℚ where star_trivial _ := rfl
-
-instance : StarOrderedRing ℚ :=
-  StarOrderedRing.ofNonnegIff (fun {_ _} => add_le_add_left) fun x => by
-    refine'
-      ⟨fun hx => _, fun hx =>
-        AddSubmonoid.closure_induction hx (by rintro - ⟨s, rfl⟩; exact mul_self_nonneg s) le_rfl
-          fun _ _ => add_nonneg⟩
-    /- If `x = p / q`, then, since `0 ≤ x`, we have `p q : ℕ`, and `p / q` is the sum of `p * q`
-      copies of `(1 / q) ^ 2`, and so `x` lies in the `AddSubmonoid` generated by square elements.
-
-      Note: it's possible to rephrase this argument as `x = (p * q) • (1 / q) ^ 2`, but this would
-      be somewhat challenging without increasing import requirements. -/
```

Comment:
I'm slightly worried that this PR has increased import requirements in general, but maybe we don't care

I'll let @j-loreaux decide.

Reply:
I think we don't care too much about this. I don't think we're using these `Star` instances (especially not the `StarOrderedRing` bits) very early in the hierarchy, are we? In fact, I haven't been aware of anything using this instance on `ℚ`. @YaelDillies, do you have plans to use this in many places?

Reply:
I'm going to use those in discrete Fourier analysis, so that shouldn't be a problem